codeGenerator/src/robotConfiguration/robotConfig.cs

```diff
@@ -89,26 +90,56 @@ robotVariants loadRobotConfiguration(string fullPathName)
                 using (var myFileStream = new FileStream(mechanismFullPath, FileMode.Open))
                 {
                     theRobotVariants.mechanism[m] = (mechanism)mySerializer.Deserialize(myFileStream);
+                    myFileStream.Close();
                 }
             }
 
-            foreach (robot theRobot in theRobotVariants.robot)
+            //try loading any additional mechanisms in cofiguration directory
+            string[] files = Directory.GetFiles(Path.GetDirectoryName(fullPathName), "*.xml");
+            foreach (string file in files)
             {
-                foreach (mechanismInstance mi in theRobot.mechanismInstance)
+                if (theRobotVariants.mechanism.Any(p => p.name == Path.GetFileName(file).Replace(".xml", "")))
```

Comment:
Instead of Path.GetFileName(file).Replace(".xml", "")) you could use GetFilenameWithoutExtension